shared/chat/conversation/info-panel/notifications/container.js

```diff
@@ -66,86 +132,17 @@ const mapStateToProps = (state: TypedState): StateProps => {
-  _onMuteConversation: (conversationIDKey: Types.ConversationIDKey, muted: boolean) => void,
-  _onSetNotification: (
-    conversationIDKey: Types.ConversationIDKey,
-    deviceType: DeviceType,
-    notifyType: Types.NotifyType
-  ) => void,
-  _onToggleChannelWide: (conversationIDKey: Types.ConversationIDKey) => void,
-}
-
-const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
-  _resetSaveState: (conversationIDKey: Types.ConversationIDKey) =>
-    dispatch(ChatGen.createSetNotificationSaveState({conversationIDKey, saveState: 'unsaved'})),
-  _onMuteConversation: (conversationIDKey: Types.ConversationIDKey, muted: boolean) => {
-    dispatch(ChatGen.createMuteConversation({conversationIDKey, muted}))
-  },
-  _onSetNotification: (
-    conversationIDKey: Types.ConversationIDKey,
-    deviceType: DeviceType,
-    notifyType: Types.NotifyType
-  ) => dispatch(ChatGen.createSetNotifications({conversationIDKey, deviceType, notifyType})),
-  _onToggleChannelWide: (conversationIDKey: Types.ConversationIDKey) =>
-    dispatch(ChatGen.createToggleChannelWideNotifications({conversationIDKey})),
-})
-
-// Use _resetSaveState as a signal for whether StateProps is empty.
-type _Props =
-  | (Props & {
-      _resetSaveState: () => void,
-    })
-  | {_resetSaveState: void}
-
-const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps): _Props => {
-  if (!stateProps.conversationIDKey) {
-    return {_resetSaveState: undefined}
-  }
-
-  const convKey = stateProps.conversationIDKey
-  return {
-    _resetSaveState: () => dispatchProps._resetSaveState(convKey),
-    channelWide: stateProps.channelWide,
-    desktop: stateProps.desktop,
-    mobile: stateProps.mobile,
-    muted: stateProps.muted,
-    saveState: stateProps.saveState,
-    onMuteConversation: (muted: boolean) => {
-      dispatchProps._onMuteConversation(convKey, muted)
-    },
-    onSetDesktop: (notifyType: Types.NotifyType) => {
-      dispatchProps._onSetNotification(convKey, 'desktop', notifyType)
-    },
-    onSetMobile: (notifyType: Types.NotifyType) => {
-      dispatchProps._onSetNotification(convKey, 'mobile', notifyType)
-    },
-    onToggleChannelWide: () => {
-      dispatchProps._onToggleChannelWide(convKey)
-    },
-  }
+  }: OwnProps)
 }
 
-class _Notifications extends React.PureComponent<_Props> {
-  componentDidMount() {
-    if (!this.props._resetSaveState) {
-      return
-    }
-
-    this.props._resetSaveState()
-  }
-
+class _MaybeNotifications extends React.PureComponent<StateProps> {
   render() {
-    if (!this.props._resetSaveState) {
+    if (!this.props.conversationID) {
       return null
     }
 
-    return <Notifications {...this.props} />
+    return <NotificationsWithKey {...this.props} />
   }
 }
 
-export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(_Notifications)
+export default connect(mapStateToProps, null, stateProps => stateProps)(_MaybeNotifications)
```

Comment:
need the dummy `mergeProps` to make flow happy

Reply:
You don't. Just explicitly type the return of mapstatetoprops as :*

Reply:
After removing the dummy `mergeProps`, replacing `: StateProps` with `: *` yields the same flow error for me